ui/pages/institutional/confirm-add-institutional-feature/confirm-add-institutional-feature.js

```diff
@@ -0,0 +1,191 @@
+import React, { useState, useContext } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import Button from '../../../components/ui/button';
+import PulseLoader from '../../../components/ui/pulse-loader';
+import { INSTITUTIONAL_FEATURES_DONE_ROUTE } from '../../../helpers/constants/routes';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import { getMostRecentOverviewPage } from '../../../ducks/history/history';
+import { Text } from '../../../components/component-library';
+import {
+  TextColor,
+  TextVariant,
+  OVERFLOW_WRAP,
+  TEXT_ALIGN,
+} from '../../../helpers/constants/design-system';
+import Box from '../../../components/ui/box';
+import { mmiActionsFactory } from '../../../store/institutional/institution-background';
+
+export default function ConfirmAddInstitutionalFeature({ history }) {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+  const mmiActions = mmiActionsFactory();
+  const [isLoading, setIsLoading] = useState(false);
+  const [connectError, setConnectError] = useState('');
+  const mostRecentOverviewPage = useSelector(getMostRecentOverviewPage);
+  const connectRequests = useSelector(
+    (state) => state.metamask.institutionalFeatures?.connectRequests,
+  );
+
+  const trackEvent = useContext(MetaMetricsContext);
+  const connectRequest = connectRequests[0];
+
+  if (!connectRequest) {
+    history.push(mostRecentOverviewPage);
+    return null;
+  }
```

Comment:
Fixed

Reply:
@brad-decker, we don’t need a useEffect for that because the behavior we want is that when there aren’t connectRequests, we return null. We only return the component when there are connectRequests

Reply:
That doesn't change the fact that the pushing of a new route shouldn't happen in the body of a component but needs to happen in an effect. I didn't do an exhaustive search in our code but I looked for every invocation where we route to mostRecentOverviewPage and also a few hot spots that I knew did something with routing as a result of the component being rendered (aka no action taken). In all cases the routing happens in a old style component lifecycle method, such as componentDidUpdate, or inside a hook or callback function that is triggered on user action. All of these are properly managed by React's effect system -- but doing the routing in the body of the function can lead to issues.

You can do **both**

have an effect that re-routes and a return null statement.

Reply:
Fixed